### PR TITLE
feat: support local filesystem

### LINKED ISSUE #344: 图片、文件、视频能否支持本地存储？
### Is your feature request related to a problem?

需求点：
- 现在图片或文件都放数据库里加载出来有点慢，每次刷新页面都会重新从数据库拉取一次文件，无法像静态资源那样缓存在浏览器
- 私人照片不想放图床
- 视频有的比较大，也不方便用网络存储

### Describe the solution you'd like

- 支持本地存储
- 支持存视频文件，视频播放支持断点续传
- 本地图片自动生成缩略图，小图使用缩略图，点击再加载原图

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Zeng1998**
I merge the commits of front-end and back-end since there are no big changes.

Close #344 #1163 #959 .

### CODE REVIEW

**Comment by HappyZ** [comment]
How do you migrate the data between local and S3 storage?

**Comment by Zeng1998** [comment]
> How do you migrate the data between local and S3 storage?

We don't need to do that.

File: store/resource.go [hunk 1]
```diff
@@ -295,7 +297,7 @@ func (s *Store) findResourceListImpl(ctx context.Context, tx *sql.Tx, find *api.
 		fields = append(fields, "blob")
 	}
 	if s.profile.IsDev() {
-		fields = append(fields, "visibility")
+		fields = append(fields, "visibility", "Internal_path")
```
Comment (boojack) [inline store/resource.go:300]:
```suggestion
		fields = append(fields, "visibility", "internal_path")
```

**Review by boojack — COMMENTED** [review]
And please fix the conflicts

File: api/system_setting.go [hunk 2]
```diff
@@ -148,6 +160,12 @@ func (upsert SystemSettingUpsert) Validate() error {
 		if err != nil {
 			return fmt.Errorf("failed to unmarshal system setting openai api config value")
 		}
+	} else if upsert.Name == SystemSettingLocalStoragePath {
+		value := ""
+		err := json.Unmarshal([]byte(upsert.Value), &value)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal system setting openai api host value")
```
Comment (boojack) [inline api/system_setting.go:167]: unrelated error message

**Review by boojack — CHANGES_REQUESTED** [review]
There are some irrelevant changes and error messages, please check throughout

**Comment by Zeng1998** [comment]
Close and re-PR when I have enough time to do it.

**Pull request closed without merging**